rust/hyperlane-base/src/settings/chains.rs

```diff
@@ -208,6 +239,8 @@ pub struct RawChainConf {
     metrics_conf: Option<PrometheusMiddlewareConf>,
     #[serde(default)]
     index: Option<RawIndexSettings>,
+    #[serde(default)]
+    sealevel: Option<RawSealevelConf>,
```

Comment:
I don't feel *super* strongly but I think leaking protocol-specific things into chain agnostic parts of the code like this is something we should avoid if possible

Originally I was thinking we'd just set `addresses.interchainGasPaymaster` to like `"${IGP program ID}-${relayer IGP account}"` but that's pretty confusing and also conflicts with the way we parse addresses / pubkeys, so I get why you avoided that

I have another idea where we can just specify the "relayer IGP account" and from there derive the IGP program ID. So we'd be able to remove this and just set the value of what's specified here in the existing `addresses.interchainGasPaymaster` place

We can make use of the fact that the IGP program ID is the owner of the "relayer IGP account", which is just a PDA.

So in `interchain_gas.rs` we can have a one-time step to get the IGP program ID by doing something like:
```
let igp_account = self
            .rpc_client
            .get_account_with_commitment(&igp_account_pubkey, CommitmentConfig::finalized())
            .await?
            .value;
let igp_program_id = igp_account.owner;
```

And we can store this in a OnceCell or something, or maybe just an Option and we have a function like `async fn get_igp_program_id() -> &Pubkey` or something like that that interior to it has logic to either use the cached version or actually make the RPC to get it

Curious what you think

Reply:
> leaking protocol-specific things into chain agnostic parts of the code like this is something we should avoid if possible

I agree in general, though if there are environment-specific settings I'd rather have them as separate optional config fields than make counterintuitive changes to the other agnostic fields. 

> Curious what you think

Your suggestion is great since we're reducing the setup complexity, so I'd much rather go for that.

If I have the IGP account as an `H256`, would I build the pubkey with `Pubkey::from(<[u8; 32]>::from(igp_account_h256))`?